### DISCUSSION TITLE
Stop build if one executable() target fails

### DISCUSSION THREAD

**Discussion by Phil67ago** [mesonbuild/meson#11962] [Q&A] [answered]
In this test, A and B are built one after the other.
How can I stop ninja from building B if A fails.

[meson.build](https://github.com/mesonbuild/meson/files/12001523/meson.build.txt)

**Comment by eli-schwartz** [comment]
This should already be the case. `ninja` (and `make` too) will error out if one target fails to build, unless you pass the `-k0`.

**Reply by Phil67ago** [reply]
I my case ninja builds A and B so what could be wrong?
ninja --version: 1.11.1
meson --version: 1.0.1
I run ninja without any arguments

**Reply by Phil67ago** [reply]
just tested exit code from cc and it is 0=ok, 1=error

**Reply by eli-schwartz** [reply, marked as answer]
One thing that could be "wrong" is that ninja could have started building B first, or at the same time, so it was already built by the time A failed.

By default ninja builds in parallel, equivalent to `make -j$(nproc)`. You can force ninja to build only one thing at a time with `ninja -j1` but it's not obvious that that's better than just failing as soon as one target fails to build.

**Reply by Phil67ago** [reply]
Your right ninja -j1 only build A!
Thanks!!!
